File: iosgl/iosgl/bt/filemode/GroupDataManager.m
//
//  GroupDataManager.m
//  iosgl
//
//  Created by zhao on 22/2/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "GroupDataManager.h"
#import "GroupRes.h"
#import "GL_Header.h"

static GroupDataManager *instance = nil;
@implementation GroupDataManager
+ (instancetype)default{
    if (instance == nil) {
        instance = [[GroupDataManager alloc] init];
    }
    return instance;
}
- (instancetype)init
{
    self = [super init];
    if (self) {
        self.dic=[[NSMutableDictionary alloc]init];
    }
    return self;
}
-(void)getGroupData:(NSString*)url Block:(void (^)(GroupRes * ))block;
{
    
    if(self.dic[url]){
        block(self.dic[url]);
    }else{
        GroupRes *groupRes=[[GroupRes alloc]init];
        [groupRes load:url  Block:^(NSString* value) {
            self.dic[url]=groupRes;
            block(self.dic[url]);
        }];
    }
}
  
@end

File: test/four/app/src/main/java/com/z3d/filemodel/GroupDataManager.java
package com.z3d.filemodel;


import com.z3d.base.CallBackFun;
import com.z3d.base.GroupBackFun;
import com.z3d.base.ResGC;
import com.z3d.base.Scene_data;
import com.z3d.res.GroupRes;
import com.z3d.scene.Scene3D;

public class GroupDataManager extends ResGC {


    public GroupDataManager(Scene3D val) {
        super(val);
    }

    public void getGroupData(final String url, final GroupBackFun bfun)
    {

//        if(self.dic[url]){
//            block(self.dic[url]);
//        }else{
//            GroupRes *groupRes=[[GroupRes alloc]init];
//        [groupRes load:url  Block:^(NSString* value) {
//                self.dic[url]=groupRes;
//                block(self.dic[url]);
//            }];
//        }

        if(this.dic.containsKey(url)){
            bfun.Bfun((GroupRes)this.dic.get(url));
        }else{
            final GroupRes groupRes=new GroupRes( scene3D);
            groupRes.load(Scene_data.fileRoot + url, new CallBackFun() {
                @Override
                public void StateChange(boolean State) {

                    dic.put(url,groupRes);
                    bfun.Bfun((GroupRes)dic.get(url));


                }
            });
        }


    }



}
